saleor/graphql/product/dataloaders/attributes.py

```diff
@@ -247,85 +167,64 @@ def map_assignment_to_values(values):
         )
 
 
-class SelectedAttributesByProductIdLoader(DataLoader):
-    context_key = "selectedattributes_by_product"
+class AttributeValuesByProductIdLoader(DataLoader):
+    context_key = "attributevalues_by_productid"
 
     def batch_load(self, keys):
-        def with_products_and_assigned_attributes(result):
-            products, product_attributes = result
-            products = [product for product in products if product is not None]
-            assigned_product_attribute_ids = [
-                a.id for attrs in product_attributes for a in attrs
-            ]
-            product_type_ids = list({p.product_type_id for p in products})
-            product_attributes = dict(zip(keys, product_attributes))
+        attribute_values = list(
+            AssignedProductAttributeValue.objects.using(self.database_connection_name)
+            .filter(product_id__in=keys)
+            .prefetch_related("value")
+            .iterator()
```

Comment:
Yes, db `iterator()` is a bit different to python iterator and using it will in this case cause the db to not store the resultset in memory, which will result in a slight optimisation at the db level. I will add a comment about it 🙌